### PR TITLE
MULE-15632: Parameterize HTTP server's path tree structure

### PR DESCRIPTION
**Pull request by afelisatti**

### CODE REVIEW

File: modules/http-api/src/main/java/org/mule/runtime/http/api/server/AcceptsAllMethodsRequestMatcher.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+package org.mule.runtime.http.api.server;
+
+import static java.lang.System.identityHashCode;
+import static java.util.Collections.singletonList;
+import org.mule.runtime.http.api.domain.message.request.HttpRequest;
+
+/**
+ * Accepts all the HTTP methods.
+ */
+class AcceptsAllMethodsRequestMatcher extends DefaultMethodRequestMatcher {
```
Comment (marianogonzalez) [inline modules/http-api/src/main/java/org/mule/runtime/http/api/server/AcceptsAllMethodsRequestMatcher.java]: aren't all this matchers already present on the other PR?

File: modules/http-api/src/main/java/org/mule/runtime/http/api/server/MethodRequestMatcherBuilder.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.Collection;
+import java.util.LinkedList;
+
+/**
+ * Builder of {@link MethodRequestMatcher}. At the very least one HTTP method must be added and instances can only be obtained via
+ * {@link MethodRequestMatcher#builder()} or {@link MethodRequestMatcher#builder(Collection)}.
+ *
+ * @since 4.1.5
+ */
+public class MethodRequestMatcherBuilder {
```
Comment (marianogonzalez) [inline modules/http-api/src/main/java/org/mule/runtime/http/api/server/MethodRequestMatcherBuilder.java]: isn't this on the other PR as well?

File: modules/http-api/src/main/java/org/mule/runtime/http/api/server/AcceptsAllMethodsRequestMatcher.java (hunk 1 above)
Comment (afelisatti) [reply modules/http-api/src/main/java/org/mule/runtime/http/api/server/AcceptsAllMethodsRequestMatcher.java]: They are deleted on the other PR.

File: modules/http-api/src/main/java/org/mule/runtime/http/api/server/MethodRequestMatcherBuilder.java (hunk 2 above)
Comment (afelisatti) [reply modules/http-api/src/main/java/org/mule/runtime/http/api/server/MethodRequestMatcherBuilder.java]: Same, it's deleted on the other.

**Pull request merged**